From 641749b89ae642e89a5b7c7f2abbfa2ac4eb0242 Mon Sep 17 00:00:00 2001
From: Wei-Luan Wang <weiluanwang@google.com>
Date: Wed, 22 Oct 2025 02:01:24 +0000
Subject: [PATCH] Revert "Pass Bus::Options by value with std::move."

This reverts commit 533382801f435c6732e89bc153dd88aa02340742.

Manually minimize the revert.

Change-Id: I7c6fb65dc872e810217aeda325494df3717daac5
---
 dbus/bus.cc      | 2 +-
 dbus/bus.h       | 2 +-
 dbus/mock_bus.cc | 3 ++-
 dbus/mock_bus.h  | 2 +-
 4 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/dbus/bus.cc b/dbus/bus.cc
index cf3b8ecacb..6ecbb3b303 100644
--- a/dbus/bus.cc
+++ b/dbus/bus.cc
@@ -166,7 +166,7 @@ Bus::Options::Options(Bus::Options&&) = default;
 
 Bus::Options& Bus::Options::operator=(Bus::Options&&) = default;
 
-Bus::Bus(Options options)
+Bus::Bus(const Options& options)
     : bus_type_(options.bus_type),
       connection_type_(options.connection_type),
       dbus_task_runner_(options.dbus_task_runner),
diff --git a/dbus/bus.h b/dbus/bus.h
index a3f4fdcb0d..81569222ca 100644
--- a/dbus/bus.h
+++ b/dbus/bus.h
@@ -226,7 +226,7 @@ class CHROME_DBUS_EXPORT Bus : public base::RefCountedThreadSafe<Bus> {
 
   // Creates a Bus object. The actual connection will be established when
   // Connect() is called.
-  explicit Bus(Options options);
+  explicit Bus(const Options& options);
 
   Bus(const Bus&) = delete;
   Bus& operator=(const Bus&) = delete;
diff --git a/dbus/mock_bus.cc b/dbus/mock_bus.cc
index 57c6d351fa..ddbb7cacba 100644
--- a/dbus/mock_bus.cc
+++ b/dbus/mock_bus.cc
@@ -10,7 +10,8 @@
 
 namespace dbus {
 
-MockBus::MockBus(Bus::Options options) : Bus(std::move(options)) {}
+MockBus::MockBus(const Bus::Options& options) : Bus(options) {
+}
 
 MockBus::~MockBus() = default;
 
diff --git a/dbus/mock_bus.h b/dbus/mock_bus.h
index 4dfd053e90..4f9aaef516 100644
--- a/dbus/mock_bus.h
+++ b/dbus/mock_bus.h
@@ -21,7 +21,7 @@ namespace dbus {
 // D-Bus calls.
 class MockBus : public Bus {
  public:
-  explicit MockBus(Bus::Options options);
+  MockBus(const Bus::Options& options);
 
   MOCK_METHOD2(GetObjectProxy, ObjectProxy*(std::string_view service_name,
                                             const ObjectPath& object_path));
-- 
2.51.0.915.g61a8936c21-goog

